crates/sui-json-rpc-types/src/sui_transaction.rs

```diff
@@ -237,6 +238,39 @@ impl PartialEq for SuiTransactionBlockResponse {
     }
 }
 
+pub fn get_new_package_obj_from_response(
+    response: &SuiTransactionBlockResponse,
+) -> Option<ObjectRef> {
+    response.object_changes.as_ref().and_then(|changes| {
+        changes
+            .iter()
+            .find(|change| matches!(change, ObjectChange::Published { .. }))
+            .map(|change| change.object_ref())
+    })
+}
+
+pub fn get_new_package_upgrade_cap_from_response(
```

Comment:
Is there any reason these are not functions of SuiTransactionBlockResponse?

Reply:
These feel too much like utility functions to be part of the class/type.

Reply:
Well, but it takes only one argument which is the class type.

Reply:
The challenges of introduce utility functions is always that they are difficult to discover.